pkg/whip/whip_handler.go

```diff
@@ -432,22 +469,54 @@ func streamKindFromCodecType(typ webrtc.RTPCodecType) types.StreamKind {
 	}
 }
 
-func validateOfferAndGetExpectedTrackCount(offer *webrtc.SessionDescription) (int, error) {
+func (h *whipHandler) validateOfferAndGetExpectedTrackCount(offer *webrtc.SessionDescription) (int, error) {
 	parsed, err := offer.Unmarshal()
 	if err != nil {
 		return 0, err
 	}
 
-	mediaTypes := make(map[string]struct{})
+	audioCount, videoCount := 0, 0
+
 	for _, m := range parsed.MediaDescriptions {
-		if _, ok := mediaTypes[m.MediaName.Media]; ok {
+		if types.StreamKind(m.MediaName.Media) == types.Audio {
 			// Duplicate track for a given type. Forbidden by the RFC
-			return 0, errors.ErrDuplicateTrack
+			if audioCount != 0 {
+				return 0, errors.ErrDuplicateTrack
+			}
+
+			audioCount++
+
+		} else if types.StreamKind(m.MediaName.Media) == types.Video {
+			// Duplicate track for a given type. Forbidden by the RFC
+			if videoCount != 0 {
+				return 0, errors.ErrDuplicateTrack
+			}
+
+			for _, a := range m.Attributes {
+				if a.Key == "simulcast" {
+					spaceSplit := strings.Split(a.Value, " ")
+					if len(spaceSplit) != 2 || spaceSplit[0] != "send" {
+						return 0, errors.ErrInvalidSimulcast
+					}
+
+					layersSplit := strings.Split(spaceSplit[1], ";")
+					if len(layersSplit) != 3 {
```

Comment:
Any reason to only support 1 or 3 simulcast tracks (and not 2)? I could imagine 2 simulcast tracks making sense if the source media is small.

Reply:
Fixed!